superset/views/core.py

```diff
@@ -889,11 +889,10 @@ def explore(
         except (SupersetException, SQLAlchemyError):
             datasource_data = dummy_datasource_data
 
-        columns: List[Dict[str, Any]] = []
         if datasource:
             datasource_data["owners"] = datasource.owners_data
             if isinstance(datasource, Query):
-                columns = datasource.extra.get("columns", [])
+                datasource_data["columns"] = datasource.extra.get("columns", [])
```

Comment:
would it be easier to make a `columns` property of the Query model that does this?